airbyte-cdk/bulk/toolkits/load-object-storage/src/test/kotlin/io/airbyte/cdk/load/state/object_storage/ObjectStorageDestinationStateTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.state.object_storage
+
+import io.airbyte.cdk.load.MockObjectStorageClient
+import io.airbyte.cdk.load.MockPathFactory
+import io.airbyte.cdk.load.command.MockDestinationCatalogFactory
+import io.airbyte.cdk.load.state.DestinationStateManager
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import jakarta.inject.Inject
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+@MicronautTest(
+    rebuildContext = true,
+    environments =
+        [
+            "ObjectStorageDestinationStateTest",
+            "MockDestinationCatalog",
+            "MockObjectStorageClient",
+            "MockPathFactory"
+        ]
+)
+class ObjectStorageDestinationStateTest {
+    @Inject lateinit var stateManager: DestinationStateManager<ObjectStorageDestinationState>
+    @Inject lateinit var mockClient: MockObjectStorageClient
+    @Inject lateinit var pathFactory: MockPathFactory
+
+    companion object {
+        val stream1 = MockDestinationCatalogFactory.stream1
+        const val PERSISTED =
+            """{"generations":[{"generation_id":0,"objects":[{"key":"key1","part_number":0},{"key":"key2","part_number":1}]},{"generation_id":1,"objects":[{"key":"key3","part_number":0},{"key":"key4","part_number":1}]}]}"""
+    }
+
+    @Test
+    fun testBasicLifecycle() = runTest {
+        // TODO: Test fallback to generation id
```

Comment:
what fallback is this referring to?

Reply:
I meant fallback to getting the generation id from the metadata